engine/engine.go

```diff
@@ -146,10 +153,7 @@ func (e *Engine) run(ctx context.Context) {
 			reqCtx := common.WithRequestId(ctx, msg.ID)
 			e.handle(reqCtx, msg)
 		case <-ctx.Done():
-			log.Event(ctx, "halting consumer", log.INFO)
-			e.consumer.Close()
-			log.Event(ctx, "waiting for handlers", log.INFO)
-			e.wg.Wait()
+
```

Comment:
does this need to call the close method you have added?

Reply:
No the `Close` function is being called through main. This will just return and allow the from the `Start` function called in main and allow the `Close` to gracefully shutdown.